lib/policy/expressions/environment.go

```diff
@@ -2,13 +2,50 @@ package expressions
 
 import (
 	"math/rand/v2"
+	"fmt"
+	"net"
 
 	"github.com/google/cel-go/cel"
 	"github.com/google/cel-go/common/types"
 	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/common/types/traits"
 	"github.com/google/cel-go/ext"
+
+	"github.com/yl2chen/cidranger"
 )
 
+func remoteAddrInList(remoteAddr types.String, ipList traits.Lister) (bool, error) {
+
+	ipAddr := net.ParseIP(string(remoteAddr))
+	if ipAddr == nil {
+		return false, fmt.Errorf("remoteAddrInList: %s is not a valid IP address", remoteAddr)
+	}
+
+	ranger := cidranger.NewPCTrieRanger()
+	it := ipList.Iterator()
+	for it.HasNext() == types.True {
+		item := it.Next()
+		cidr, ok := item.(types.String)
+		if !ok {
+			continue
+		}
+
+		_, rng, err := net.ParseCIDR(string(cidr))
+		if err != nil {
+			return false, fmt.Errorf("remoteAddrInList: address %s CIDR parse error: %w", cidr, err)
+		}
+
+		ranger.Insert(cidranger.NewBasicRangerEntry(*rng))
+	}
```

Comment:
Is there a way to cache this? Feels like this is a poster child for data that we don't have to parse every time.

Reply:
I would simply move the codeblock of L24 to L39 into a separate function which is cached. The IP ranges should be relatively consistent with `remoteAddr` being the inconstant one. If you want some help on this PR feel free to give me a shout

Reply:
Sure, I would greatly appreciate the help -- Golang is not my thing.

I have been experimenting (proto fully functional) with non-expression rules and adding a `match-all` tag to rule definitions. Go doesn't easily lend itself to function overloading so I had to tweak the `Checker` object a bit. If you would prefer to go (_haha_) with a different solution to the same problem I can submit another PR.